src/main/java/org/hydev/mcpm/client/arguments/parsers/MirrorParser.java

```diff
@@ -0,0 +1,58 @@
+package org.hydev.mcpm.client.arguments.parsers;
+
+import net.sourceforge.argparse4j.impl.Arguments;
+import net.sourceforge.argparse4j.inf.Namespace;
+import net.sourceforge.argparse4j.inf.Subparser;
+import org.hydev.mcpm.client.commands.entries.MirrorController;
+
+import java.util.function.Consumer;
+
+/**
+ * Parser for the mirror selector command
+ *
+ * @param controller Mirror controller
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-22
+ */
+public record MirrorParser(MirrorController controller) implements CommandParser
+{
+    @Override
+    public String name()
+    {
+        return "mirror";
+    }
+
+    @Override
+    public String description()
+    {
+        return "Select a source (mirror) to download plugins from";
+    }
+
+    @Override
+    public void run(Namespace details, Consumer<String> log)
+    {
+        switch (details.getString("op"))
+        {
+            case "ping" -> controller.ping(details.getBoolean("refresh"), log);
+            case "sel" -> controller.sel(details.getString("host"), log);
```

Comment:
This is nice. I wish the method name was English though:
```suggestion
            case "sel" -> controller.select(details.getString("host"), log);
```

Reply:
ok fixed